### DISCUSSION TITLE
`brew doctor` cannot detect `Xcode: 15.2`

### DISCUSSION THREAD

**Discussion by osalbahr** [Homebrew/discussions#5676] [Tap maintenance and Homebrew development] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.4.1-34-gaf958b2
ORIGIN: https://github.com/Homebrew/brew
HEAD: af958b2540458a460081ae0426a1c4a070a40875
Last commit: 2 hours ago
Core tap JSON: 16 Oct 05:41 UTC
Core cask tap JSON: 16 Oct 05:41 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 4
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.3.5 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.3.5/bin/ruby
CPU: quad-core 64-bit kabylake
Clang: 14.0.3 build 1403
Git: 2.39.2 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 13.7-x86_64
CLT: 14.3.1.0.1.1683849156
Xcode: 15.2 => /Users/adminaccount/Downloads/Xcode.app/Contents/Developer
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: A newer Command Line Tools release is available.
Update them from Software Update in System Settings.

If that doesn't show you any updates, run:
  sudo rm -rf /Library/Developer/CommandLineTools
  sudo xcode-select --install

Alternatively, manually download them from:
  https://developer.apple.com/download/all/.
You should download the Command Line Tools for Xcode 15.2.
```

### Description of issue

As said above, I can just ignore the message. But I'd like it if `brew dr` did not report on this false negative (Xcode does exist, but can't be detected).

For reference, I installed Xcode 15.2 using https://download.developer.apple.com/Developer_Tools/Xcode_15.2/Xcode_15.2.xip.

Update: Installing [Command Line Tools for Xcode 15.1](https://download.developer.apple.com/Developer_Tools/Command_Line_Tools_for_Xcode_15.1/Command_Line_Tools_for_Xcode_15.1.dmg) fixed it.

**Comment by SMillerDev** [comment]
It doesn't say anything about XCode. It mentions CommandLine Tools, which is a different piece of software

**Reply by osalbahr** [reply]
Good catch. I confused the two. However I do have CommandLine Tools:

```console
% git --version
git version 2.39.2 (Apple Git-143)
```

But perhaps `CLT: 14.3.1.0.1.1683849156` isn't latest. Will look into it. Thank you!

**Comment by osalbahr** [comment]
It seems that [Apple](https://developer.apple.com/download/all) does not provide CL Tools for Xcode 15.2, not sure what the proper workaround is.

Edit: the best workaround is probably upgrading to Xcode 15.3.

<img width="1050" alt="Screenshot 1446-04-13 at 10 02 23 AM" src="https://github.com/user-attachments/assets/ee1af6ae-c09c-4103-a824-c99da03d052f">

**Reply by osalbahr** [reply]
If anyone is having the same issue, it seems that Apple Xcode pulled the plug on Ventura. I can't install the latest in the App Store. So I guess I have to find the latest supported version I can install that has a corresponding CL Tools.

<img width="1292" alt="Screenshot 1446-04-13 at 10 06 59 AM" src="https://github.com/user-attachments/assets/38dc20f0-aaed-4d6e-9f1a-62d431c17e64">

**Comment by osalbahr** [comment, marked as answer]
Update: installing [Command Line Tools for Xcode 15.1](https://download.developer.apple.com/Developer_Tools/Command_Line_Tools_for_Xcode_15.1/Command_Line_Tools_for_Xcode_15.1.dmg) fixed it.

```console
% brew dr
Your system is ready to brew.
```
```console
% brew config
HOMEBREW_VERSION: 4.4.1-34-gaf958b2
ORIGIN: https://github.com/Homebrew/brew
HEAD: af958b2540458a460081ae0426a1c4a070a40875
Last commit: 5 hours ago
Core tap JSON: 16 Oct 05:50 UTC
Core cask tap JSON: 16 Oct 05:50 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 4
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.3.5 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.3.5/bin/ruby
CPU: quad-core 64-bit kabylake
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 13.7-x86_64
CLT: 15.1.0.0.1.1700200546
Xcode: 15.2 => /Users/adminaccount/Downloads/Xcode.app/Contents/Developer
```

**Reply by erikw** [reply]
I can confirm, this worked for me too. Homebrew wanted to have `15.2` but installing `15.1` manually made `brew doctor` to stop complaining.

**Reply by osalbahr** [reply]
Glad it worked! But FYI I currently use Xcode 26.0 (Beta) manually installed from the website.
